### PR TITLE
feat(adherence+stock): heatmap + timeline (I02, EV-07)

### PR DESCRIPTION
**Pull request by coelhotv**
## Summary

**Sprint 6.5** implements two key insights from Fase 6:

- **I02: AdherenceHeatmap** — 7×4 grid (days × time periods) analyzing adherence patterns with 21+ days of data
- **EV-07: PrescriptionTimeline** — Visual timeline of prescription validity with status indicators

Both features use pure client-side logic (no extra Supabase calls) with full Zod validation.

### I02: AdherenceHeatmap (8 SP)
- **Service**: `adherencePatternService.js` (171 lines, pure function)
  - Validates input via `AnalyzeAdherencePatternsInputSchema` (Zod)
  - Pre-processes protocols to map expected doses per day/period
  - Counts actual doses taken from logs
  - Calculates adherence %= min(100%, taken/expected × 100)
  - Identifies worst cell (minimum 3 expected doses threshold)
  - Generates narrative: "Seu pior horário é {day} à {period}"

- **Component**: `AdherenceHeatmap.jsx` (150 lines, responsive)
  - Desktop (≥380px): 7×4 grid with opacity-based coloring
  - Mobile (<380px): Stacked cards with bar-fills
  - Tooltip on hover/tap showing adherence %
  - Accessibility: aria-label, role="gridcell", keyboard support
  - Feature flag: Only renders with ≥21 days of data

- **Integration**: `HealthHistory.jsx`
  - Added memo `adherencePatternData` to calculate pattern from timelineLogs + active protocols
  - Component rendered below calendar in glass-card

### EV-07: PrescriptionTimeline (3 SP)
- **Component**: Already exists at `src/features/stock/components/PrescriptionTimeline.jsx`
- **Integration**: `Stock.jsx`
  - Added memo `prescriptionTimelineData` to process active protocols
  - Status calculation: VIGENTE (>30d), VENCENDO (7-30d), VENCIDA (expired)
  - Renders timeline section above "Custo Mensal"

## Quality Gate ✅

- **Tests**: 526/526 passing (includes 14 new tests for adherencePatternService)
- **Linting**: 0 errors
- **Coverage**: 100% happy path + edge cases (< 21 days, all 100%, etc.)
- **Zod Validation**: Required on all inputs (schema.safeParse())
- **Responsive**: Mobile-first with media queries
- **Accessibility**: WCAG 2.1 AA (aria-label, role, keyboard nav, 44px touch targets)

## Spec References
- Section 9: I02 AdherenceHeatmap (8 SP)
- Section 9: EV-07 PrescriptionTimeline (3 SP)
- Section 9.5: UI wireframes
- Section 9.6: Accessibility requirements

## Testing Checklist
- [x] npm run validate:agent (526 tests, 0 lint errors)
- [x] Manual: Heatmap renders with ≥21 days of data
- [x] Manual: Mobile responsive (<380px stacked layout)
- [x] Manual: PrescriptionTimeline visible in Stock view
- [x] Code review ready for Gemini Code Assist

🤖 Generated with Claude Code

### CODE REVIEW

File: src/features/adherence/services/adherencePatternService.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      const taken = grid[dayIndex][periodIndex].taken
+
+      grid[dayIndex][periodIndex].expected = expected
+
+      if (expected > 0) {
+        grid[dayIndex][periodIndex].adherence = Math.min(100, Math.round((taken / expected) * 100))
+      } else {
+        // Se não há doses esperadas neste período, considerar como 100%
+        grid[dayIndex][periodIndex].adherence = 100
+      }
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply src/features/adherence/services/adherencePatternService.js]: correção aplicada no commit 7d51fa8
@gemini-code-assist pode validar

File: src/features/adherence/services/adherencePatternService.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply src/features/adherence/services/adherencePatternService.js]: correção aplicada no commit 7d51fa8
@gemini-code-assist pode validar

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #321 para tracking desta sugestão de refactoring.

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #327 para tracking desta sugestão de refactoring.

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #330 para tracking desta sugestão de refactoring.
